packages/aws-cdk-lib/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -88,6 +90,21 @@ export interface BucketDeploymentProps {
    */
   readonly prune?: boolean;
 
+  /**
+   * Suppress the warning about explicit prune setting requirement.
+   *
+   * This property serves as an escape hatch when the feature flag `@aws-cdk/aws-s3-deployment:requireExplicitPrune`
+   * is enabled. When the flag is enabled, you must either:
+   * - Explicitly set `prune: true` or `prune: false`, OR
+   * - Set `suppressPruneWarning: true` to use the default behavior
+   *
+   * This ensures you make a conscious decision about whether files in the destination bucket
+   * should be deleted when they don't exist in the source.
+   *
+   * @default false
+   */
+  readonly suppressPruneWarning?: boolean;
```

Comment:
this should not be its own property on the construct. if you use `Annotations.addWarningV2`, then there is a built in way of acknowledging the warning that the user can specify: `Annotations.acknowledgeWarning`.